test/unit/providers/portion/PortionProvider.test.ts

```diff
@@ -0,0 +1,253 @@
+        // You can optionally mock other methods here, such as post, put, etc.
+      };
+      createSpy.mockReturnValueOnce(axiosInstanceMock);
+
+      const portionCache = new NodeCache({ stdTTL: 600 });
+      const portionFetcher = new PortionFetcher('https://portion.uniswap.org/', portionCache);
+      const tokenFetcher = new TokenFetcher();
+      const portionProvider = new DefaultPortionProvider(portionFetcher, tokenFetcher);
+
+      const allPairs: Array<Array<BaseCurrency>> = [
+        [Ether.onChain(ChainId.MAINNET), USDC_ON(ChainId.MAINNET)],
+        [WETH9[ChainId.MAINNET], USDT_ON(ChainId.MAINNET)],
+        [DAI_ON(ChainId.MAINNET), WBTC_MAINNET],
+        [BUSD_MAINNET, EUROC_MAINNET],
+        [GUSD_MAINNET, LUSD_MAINNET],
+        // [agEUR_MAINNET, XSGD_MAINNET], TODO: add agEUR_MAINNET and XSGD_MAINNET into default token list
+      ]
+
+      allPairs.forEach((pair) => {
+        const token1: Currency | Token = pair[0].isNative ? pair[0] as Currency : pair[0].wrapped
+        const token2: Currency | Token = pair[1].isNative ? pair[1] as Currency : pair[1].wrapped
+        const tokenSymbol1 = token1.symbol!
+        const tokenSymbol2 = token2.symbol!
+        const tokenAddress1 = token1.wrapped.address
+        const tokenAddress2 = token2.wrapped.address
+
+        it(`token address ${tokenAddress1} to token address ${tokenAddress2} within the list, should have portion`, async () => {
+          const requestedAmount = BigNumber.from(parseAmount('1.01', token1).quotient.toString())
+          const sharedInfo: QuoteRequestInfo = {
+            requestId: uuid(),
+            tokenInChainId: ChainId.MAINNET,
+            tokenOutChainId: ChainId.MAINNET,
+            tokenIn: tokenAddress1,
+            tokenOut: tokenAddress2,
+            amount: requestedAmount,
+            type: TradeType.EXACT_INPUT,
+            slippageTolerance: "0.5",
+            swapper: uuid(),
+            useUniswapX: false,
+          }
+          await exactInGetPortionAndAssert(sharedInfo, token2)
+        });
+
+        it(`token symbol ${tokenSymbol1} to token symbol ${tokenSymbol2} within the list, should have portion`, async () => {
+          const requestedAmount = BigNumber.from(parseAmount('1.01', token1).quotient.toString())
+          const sharedInfo: QuoteRequestInfo = {
+            requestId: uuid(),
+            tokenInChainId: ChainId.MAINNET,
+            tokenOutChainId: ChainId.MAINNET,
+            tokenIn: tokenSymbol1,
+            tokenOut: tokenSymbol2,
+            amount: requestedAmount,
+            type: TradeType.EXACT_INPUT,
+            slippageTolerance: "0.5",
+            swapper: uuid(),
+            useUniswapX: false,
+          }
+
+          await exactInGetPortionAndAssert(sharedInfo, token2)
+        });
+      })
+
+      async function exactInGetPortionAndAssert(sharedInfo: QuoteRequestInfo, token2: Currency | Token) {
+        const quoteAmount = parseAmount('1605.56', token2);
```

Comment:
can we move this magic number into a parameter or constant, like `expectedQuote` or something